client/dashboard/app/routes/domain-routes.ts

```diff
@@ -92,11 +96,15 @@ export const domainDnsAddRoute = createRoute( {
 	)
 );
 
+// This route will also use a `recordId` query parameter to determine which record to edit
 export const domainDnsEditRoute = createRoute( {
 	getParentRoute: () => domainRoute,
 	path: 'dns/edit',
+	loader: ( { params: { domainName } } ) => {
+		return queryClient.ensureQueryData( domainDnsQuery( domainName ) );
+	},
 } ).lazy( () =>
-	import( '../../sites/domains/placeholder' ).then( ( d ) =>
+	import( '../../domains/dns/add' ).then( ( d ) =>
```

Comment:
For consistency, let's align with @hambai 's pattern in his PR https://github.com/Automattic/wp-calypso/pull/105113, where we extract the code into 3 components:

- `add.tsx`
- `edit.tsx`
- `form.tsx`

I think it's a good CRUD pattern to follow.

It's to avoid so many "is edit?" logic in the codebase, and to remove `useEffect()` when populating existing dns record. 😄

Reply:
Yeah I agreed. It would be better to have something like `DNSRecordForm` component by passing different props.